### ISSUE TITLE
Consider replacing fargate with codebuild

### ISSUE THREAD

**Issue by StevenACoffman** [mhausenblas/eksphemeral#2]
So we do something similar, but we use codebuild instead of fargate. Codebuild can also run an arbitrary container image (e.g. containing eksctl) with an arbitrary iam role (necessary for eksctl) and it will run it. You can then add a github webhook that creates a new cluster on every commit/pullrequest/etc, or you can use a lambda to initiate the cluster on a cron schedule (like 8am Monday through Friday). Not sure if this is a helpful suggestion for your use case, but we found fargate kind of awkward and expensive to use for a similar purpose.

[README.md.txt](https://github.com/mhausenblas/eksphemeral/files/3269078/README.md.txt)

[create-codebuild-project.sh.txt](https://github.com/mhausenblas/eksphemeral/files/3269077/create-codebuild-project.sh.txt)

**Comment by mhausenblas** [comment]
Thank you for your suggestion @StevenACoffman and let me think about this for a bit. My main use case is: fire up a cluster and not have to think of tearing it down. So for now ECS on Fargate does a good (enough) job. 

Can you please elaborate on the "found Fargate awkward and expensive" bit, I'm interested to learn more here :)

**Comment by StevenACoffman** [comment]
It may be very different now, but when we explored it we found that fargate was about 4 times the price of an on-demand equivalent EC2 instances. The awkwardness was probably just personal preference. The "i just want to run this bash command in the cloud" experience for fargate was not as frictionless as I'd hoped. Launching an ec2 instance via the cli and running something on it was easier.

The [jpignata](https://github.com/jpignata/fargate) tool you are using makes fargate experience much nicer, and the [turnerlabs/fargate](https://github.com/turnerlabs/fargate) fork lets you use docker compose. 

Our focus is on avoiding leaving dangling resources while providing high fidelity replica staging environments on demand, and our experience in kubernetes makes some solutions a more obvious fit. I'm sure with more ECS experience, and a different focus, a completely different solution would have been preferred. 

Thanks for putting this up and entertaining my suggestion. It's very interesting to see similar efforts from different perspectives.
